### PR TITLE
Add DetachFork and NewForkWithoutCopy

### PR DESCRIPTION
**Pull request by pleroy**
Also rename a protocol buffer.  This is the first step towards solving #972.

### CODE REVIEW

File: physics/forkable_body.hpp [hunk 1]
```diff
@@ -286,17 +286,40 @@ not_null<Tr4jectory*> Forkable<Tr4jectory, It3rator>::NewFork(
...
   child_forkable->parent_ = that();
-  child_forkable->position_in_parent_children_ = const_child_it;
+  child_forkable->position_in_parent_children_ = child_it;
   child_forkable->position_in_parent_timeline_ = timeline_it;
 
   return child_forkable.get();
 }
 
+template <typename Tr4jectory, typename It3rator>
+void Forkable<Tr4jectory, It3rator>::DetachForkWithCopiedBegin() {
```
Comment (eggrobin) [inline physics/forkable_body.hpp]: Have this return a `unique_ptr`

File: physics/forkable_body.hpp [hunk 2]
```diff
@@ -286,17 +286,40 @@ not_null<Tr4jectory*> Forkable<Tr4jectory, It3rator>::NewFork(
...
+  // ensured that now it is, so point them to the beginning of this timeline.
+  for (auto const& pair : children_) {
+    std::unique_ptr<Tr4jectory> const& child = pair.second;
+    if (child->position_in_parent_timeline_ == timeline_end()) {
+      child->position_in_parent_timeline_ = timeline_begin();
+    }
+  }
+
+  // Remove this trajectory from the children of its parent.
+  (*position_in_parent_children_)->second.release();
```
Comment (eggrobin) [inline physics/forkable_body.hpp]: Instead of releasing, move to a temporary and return move at the end.

File: physics/forkable_body.hpp (hunk 1 above)
Comment (pleroy) [inline physics/forkable_body.hpp]: Done.

File: physics/forkable_body.hpp (hunk 2 above)
Comment (pleroy) [inline physics/forkable_body.hpp]: Done.

**Pull request merged**